src/private_hack/mod.rs

```diff
@@ -146,19 +155,181 @@ impl From<LabelSelectionState> for egui::text_selection::LabelSelectionState {
 
 #[derive(serde::Deserialize, serde::Serialize)]
 pub struct Options {
-    pub style: std::sync::Arc<Style>,
+    #[serde(skip)]
+    pub dark_style: std::sync::Arc<Style>,
+    #[serde(skip)]
+    pub light_style: std::sync::Arc<Style>,
+    pub theme_preference: ThemePreference,
+    pub fallback_theme: Theme,
+    #[serde(skip)]
+    pub system_theme: Option<Theme>,
     pub zoom_factor: f32,
     pub zoom_with_keyboard: bool,
     pub tessellation_options: epaint::TessellationOptions,
     pub repaint_on_widget_change: bool,
+    pub max_passes: std::num::NonZeroUsize,
     pub screen_reader: bool,
     pub preload_font_glyphs: bool,
     pub warn_on_id_clash: bool,
     pub line_scroll_speed: f32,
     pub scroll_zoom_speed: f32,
+    pub input_options: InputOptions,
     pub reduce_texture_memory: bool,
 }
 
+impl Default for Options {
```

Comment:
Is there a reason to include these `impl` blocks? As far as I can see, the code does not call any of the functions. It might keep the `private_hack` simpler and cleaner if they were removed.